lib/improver/spotdata/read_input.py

```diff
@@ -144,14 +142,22 @@ def get_additional_diagnostics(diagnostic_name, diagnostic_data_path,
                       data, with a single entry is time_extract is provided.
 
     """
-    cubes = Load('multi_file').process(
-        diagnostic_data_path + '/*/*' + diagnostic_name + '*',
-        None)
+    # Search directory structure for all files relevant to current diagnostic.
+    files_to_read = [
```

Comment:
The base directory is diagnostic_data_path. I was worried that it was a bit naive to assume there would always be a directory hierarchy two levels deep from the base directory to find data, e.g. someone just wanting to work on some data in their home space. Is there an intermediate solution that would make you happier?